include/llfio/v2.0/detail/impl/tls_socket_sources/openssl.ipp

```diff
@@ -471,16 +471,28 @@ class openssl_socket_handle final : public tls_socket_handle
   optional<filesystem::path> _authentication_certificates_path;
   std::string _connect_hostname_port;
 
+  /* We use a registered buffer from the underlying transport for reads only, but not writes.
```

Comment:
Why mem bio? You have custom BIO openssl_custom_bio_method_t

Reply:
Thanks for your comment. Can you explain some more about what you mean? We don't use mem BIOs, the comment here refers to how OpenSSL implements i/o buffers within itself.

Reply:
If you wrap your fixed size buffers in your custom BIO you have full control over it. The growing write buffer is only a problem if you use memory BIO.

But it is very likely that I misunderstand.

Reply:
Oh I see what you mean now.

I considered implementing my own memory BIO, but from inspection of OpenSSL's implementation code, it handles failure to allocate by a memory BIO as an aborting event, not as a retry later event. I may have misread the code, I didn't do a deep dive, but from cursory inspection this seemed to be so.

Does this match your understanding?